base_tier_validation_correction/models/tier_correction.py

```diff
@@ -0,0 +1,306 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import logging
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+_logger = logging.getLogger(__name__)
+
+
+class TierCorrection(models.Model):
+    _name = "tier.correction"
+    _inherit = ["mail.thread", "mail.activity.mixin"]
+    _description = "Tier Review Correction"
+    _order = "id desc"
+
+    name = fields.Char(
+        string="Description",
+        required=True,
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+    )
+    model_id = fields.Many2one(
+        comodel_name="ir.model",
+        string="On Model",
+        domain=lambda self: [
+            (
+                "model",
+                "in",
+                self.env["tier.definition"]._get_tier_validation_model_names(),
+            )
+        ],
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+    )
+    model = fields.Char(related="model_id.model", index=True, store=True)
+    correction_type = fields.Selection(
+        selection=[
+            ("reviewer", "Reassign Reviewer(s)"),
+        ],
+        string="Correction Type",
+        default="reviewer",
+        required=True,
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+    )
+    search_name = fields.Char(
+        string="Name Search",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+    )
+    old_reviewer_ids = fields.Many2many(
+        comodel_name="res.users",
+        relation="tier_correction_old_reviewer_rel",
+        string="Original Reviewer(s)",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        help="Find documents with tier reviews matching some reviewers",
+    )
+    new_reviewer_ids = fields.Many2many(
+        comodel_name="res.users",
+        relation="tier_correction_new_reviewer_rel",
+        string="Reassign Reviewer(s)",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        help="Reassign these reviewers to the tier reviews of the found document",
+    )
+    item_ids = fields.One2many(
+        comodel_name="tier.correction.item",
+        inverse_name="correction_id",
+        readonly=True,
+        states={"draft": [("readonly", False)], "prepare": [("readonly", False)]},
+    )
+    state = fields.Selection(
+        selection=[
+            ("draft", "Draft"),
+            ("prepare", "Preparing"),
+            ("done", "Corrected"),
+            ("cancel", "Cancelled"),
+            ("revert", "Reverted"),
+        ],
+        string="Status",
+        readonly=True,
+        copy=False,
+        index=True,
+        tracking=1,
+        default="draft",
+    )
+    reference = fields.Char(
+        string="Affected Documents",
+        compute="_compute_reference",
+        store=True,
+    )
+    date_schedule_correct = fields.Datetime(
+        string="Scheduled Correct Date",
```

Comment:
Small wording problem here, "correct" is a verb or and adjective, here I think you need the noun "correction".

- "Scheduled Correct Date" sounds like this is the correct scheduled date.
- "Scheduled correction Date" sounds like the date in which the correction is going to take place. I think this is the real meaning of this field.